java/integTest/src/test/java/glide/TransactionTestUtilities.java

```diff
@@ -244,25 +246,51 @@ private static Object[] listCommands(BaseTransaction<?> transaction) {
-            5L, // llen(listKey1)
-            value3, // lindex(key5, 0)
-            1L, // lrem(listKey1, 1, value1)
-            OK, // ltrim(listKey1, 1, -1)
-            new String[] {value3, value2}, // lrange(listKey1, 0, -2)
-            value3, // lpop(listKey1)
-            new String[] {value2, value1}, // lpopCount(listKey1, 2)
-            3L, // rpush(listKey2, new String[] {value1, value2, value2})
-            value2, // rpop(listKey2)
-            new String[] {value2, value1}, // rpopCount(listKey2, 2)
-            0L, // rpushx(listKey3, new String[] { "_" })
-            0L, // lpushx(listKey3, new String[] { "_" })
-            3L, // lpush(listKey3, new String[] { value1, value2, value3})
-            4L, // linsert(listKey3, AFTER, value2, value2)
-            new String[] {listKey3, value3}, // blpop(new String[] { listKey3 }, 0.01)
-            new String[] {listKey3, value1}, // brpop(new String[] { listKey3 }, 0.01)
-        };
+        if (REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0")) {
+            transaction
+                    .lpush(listKey4, new String[] {value1, value2, value3})
+                    .lmpop(new String[] {listKey4}, LmPopDirection.LEFT)
+                    .lmpop(new String[] {listKey4}, LmPopDirection.LEFT, 2L);
+        }
+
+        var expectedResults =
+                new Object[] {
+                    5L, // lpush(listKey1, new String[] {value1, value1, value2, value3, value3})
+                    5L, // llen(listKey1)
+                    value3, // lindex(key5, 0)
+                    1L, // lrem(listKey1, 1, value1)
+                    OK, // ltrim(listKey1, 1, -1)
+                    new String[] {value3, value2}, // lrange(listKey1, 0, -2)
+                    value3, // lpop(listKey1)
+                    new String[] {value2, value1}, // lpopCount(listKey1, 2)
+                    3L, // rpush(listKey2, new String[] {value1, value2, value2})
+                    value2, // rpop(listKey2)
+                    new String[] {value2, value1}, // rpopCount(listKey2, 2)
+                    0L, // rpushx(listKey3, new String[] { "_" })
+                    0L, // lpushx(listKey3, new String[] { "_" })
+                    3L, // lpush(listKey3, new String[] { value1, value2, value3})
+                    4L, // linsert(listKey3, AFTER, value2, value2)
+                    new String[] {listKey3, value3}, // blpop(new String[] { listKey3 }, 0.01)
+                    new String[] {listKey3, value1}, // brpop(new String[] { listKey3 }, 0.01)
+                };
+
+        if (REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0")) {
+            return concatenateArrays(
+                    expectedResults,
+                    new Object[] {
+                        3L,
+                        Map.of(
+                                listKey4,
+                                new String[] {value3}), // lmpop(new String[] {listKey4},LmPopDirection.LEFT)
```

Comment:
nit: you can simplify the comment to let spotless fit this into a single line
```suggestion
                                new String[] {value3}), // lmpop(listKey4, LEFT)
```